tools/db/phone-users/src/PhoneUsers/Lib.hs

```diff
@@ -0,0 +1,172 @@
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module PhoneUsers.Lib where
+
+import Cassandra as C
+import Cassandra.Settings as C
+import Data.Conduit
+import qualified Data.Conduit.Combinators as Conduit
+import Data.Id (TeamId, UserId)
+import Data.Time
+import qualified Database.CQL.Protocol as CQL
+import Imports
+import Options.Applicative
+import PhoneUsers.Types
+import qualified System.IO as SIO
+import qualified System.Logger as Log
+import Wire.API.Team.Feature (FeatureStatus (FeatureStatusDisabled, FeatureStatusEnabled))
+import Wire.API.User (AccountStatus (Active))
+
+lookupClientsLastActiveTimestamps :: ClientState -> UserId -> IO [Maybe UTCTime]
+lookupClientsLastActiveTimestamps client u = do
+  runClient client $ runIdentity <$$> retry x1 (query selectClients (params LocalQuorum (Identity u)))
+  where
+    selectClients :: PrepQuery R (Identity UserId) (Identity (Maybe UTCTime))
+    selectClients = "SELECT last_active from clients where user = ?"
+
+readUsers :: ClientState -> ConduitM () [UserRow] IO ()
+readUsers client =
+  transPipe (runClient client) (paginateC selectUsersAll (paramsP LocalQuorum () 200) x5)
+    .| Conduit.map (fmap CQL.asRecord)
+  where
+    selectUsersAll :: C.PrepQuery C.R () (CQL.TupleType UserRow)
+    selectUsersAll =
+      "SELECT id, email, phone, activated, status, team FROM user"
+
+getConferenceCalling :: ClientState -> TeamId -> IO (Maybe FeatureStatus)
+getConferenceCalling client tid = do
+  runClient client $ runIdentity <$$> retry x1 (query1 select (params LocalQuorum (Identity tid)))
+  where
+    select :: PrepQuery R (Identity TeamId) (Identity FeatureStatus)
+    select =
+      "select conference_calling from team_features where team_id = ?"
+
+process :: Maybe Int -> ClientState -> ClientState -> IO Result
+process limit brigClient galleyClient =
+  runConduit $
+    readUsers brigClient
+      .| Conduit.mapM (\chunk -> SIO.hPutStr stderr "." $> chunk)
+      .| Conduit.concat
+      .| (maybe (Conduit.filter (const True)) Conduit.take limit)
+      .| Conduit.mapM (getUserInfo brigClient galleyClient)
+      .| Conduit.foldMap infoToResult
+
+getUserInfo :: ClientState -> ClientState -> UserRow -> IO UserInfo
+getUserInfo brigClient galleyClient ur = do
+  if not $ isCandidate
+    then pure NoPhoneUser
+    else do
+      -- should we give C* a little break here and add a small threadDelay?
+      threadDelay 200
```

Comment:
Its probably not necessary, but please run it on staging and check. 0.2 ms is barely a break anyway.